data/org.freedesktop.portal.InputCapture.xml

```diff
@@ -0,0 +1,552 @@
+        "Region".
+
+        If the zones change (e.g. a monitor is unplugged), the
+        #org.freedesktop.portal.InputCapture::ZonesChanged signal is emitted
+        and the application should re-request the current zones to update its internal state.
+
+        Note that zones are session-specific, there is no guarantee that two
+        applications see the same screen zones. An empty zone list implies
+        that no pointer barriers can be set.
+
+        Whenever the application requests the current zones, a unique serial
+        number is returned that references this set of zones. To establish
+        a pointer barrier, the application must pass this serial to
+        org.freedesktop.portal.InputCapture.SetPointerBarriers(). A serial mismatch
+        implies the application is not using the current zone set and
+        pointer barriers will fail.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>zones a(uuii)</term>
+            <listitem><para>
+              An array of retions, each specifying that zone's width,
+              height, x/y offset.
+            </para></listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>serial u</term>
+            <listitem><para>
+             A serial number to be used in the
+             org.freedesktop.portal.InputCapture.SetPointerBarriers() method to refer to
+             this set of zones. This serial increases by an unspecified
+             amount whenever the zones change and pointer barriers can only be set up
+             if the serial matches the most recent retured serial.
+
+             Note that this serial has no relation to the serial in
+             org.freedesktop.portal.InputCapture.Enable()
+            </para></listitem>
+          </varlistentry>
+        </variablelist>
+    -->
+    <method name="GetZones">
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="a{sv}" name="options" direction="in"/>
+      <arg type="o" name="handle" direction="out"/>
+    </method>
+
+    <!--
+        SetPointerBarriers:
+        @session_handle: Object path for the #org.freedesktop.portal.Session object
+        @options: Vardict with optional further information
+        @barriers: An array of vardicts, each specifying one barrier
+        @serial: A unique serial number referring to the zone  set
+        @handle: Object path for the #org.freedesktop.portal.Request object representing this call
+
+        Set up zero or more pointer barriers. Pointer barriers are horizontal
+        or vertical lines that should trigger the start of input capture when the cursor moves
+        across the pointer barrier. In the future,
+        when the compositor has deemed the pointer barrier to be crossed, it may
+        complete the org.freedesktop.portal.InputCapture.Enable() request and input events
```

Comment:
We're using normal signals instead of long-lived method calls now.

Reply:
Updated